pkg/controller/backup_session.go

```diff
@@ -391,3 +402,84 @@ func (c *StashController) getBackupSessionPhase(backupSession *api_v1beta1.Backu
 	// backup has been completed successfully
 	return api_v1beta1.BackupSessionSucceeded, nil
 }
+
+func (c *StashController) ensureBackupJobForVolumeSnapshot(backupConfig *api_v1beta1.BackupConfiguration, backupSession *api_v1beta1.BackupSession) error {
+	jobMeta := metav1.ObjectMeta{
+		Name:      VolumeSnapshotPrefix + backupSession.Name,
+		Namespace: backupSession.Namespace,
+	}
+
+	backupConfigRef, err := reference.GetReference(stash_scheme.Scheme, backupConfig)
+	if err != nil {
+		return err
+	}
+
+	backupSessionRef, err := reference.GetReference(stash_scheme.Scheme, backupSession)
+	if err != nil {
+		return err
+	}
+
+	serviceAccountName := "default"
+	//ensure respective RBAC stuffs
+	//Create new ServiceAccount
+	serviceAccountName = backupConfig.Name
+	saMeta := metav1.ObjectMeta{
+		Name:      serviceAccountName,
+		Namespace: backupConfig.Namespace,
+	}
+	_, _, err = core_util.CreateOrPatchServiceAccount(c.kubeClient, saMeta, func(in *core.ServiceAccount) *core.ServiceAccount {
+		core_util.EnsureOwnerReference(&in.ObjectMeta, backupConfigRef)
+		if in.Labels == nil {
+			in.Labels = map[string]string{}
+		}
+		in.Labels[util.LabelApp] = util.AppLabelStash
+		return in
+	})
+	if err != nil {
+		return err
+	}
+
+	err = c.ensureVolumeSnapshotJobRBAC(backupConfigRef, serviceAccountName)
+	if err != nil {
+		return err
+	}
+
+	image := docker.Docker{
+		Registry: c.DockerRegistry,
+		Image:    docker.ImageStash,
+		Tag:      c.StashImageTag,
+	}
+
+	// create Snapshot Volume Job
+	_, _, err = batch_util.CreateOrPatchJob(c.kubeClient, jobMeta, func(in *batchv1.Job) *batchv1.Job {
+		// set BackupSession as owner of this Job
+		core_util.EnsureOwnerReference(&in.ObjectMeta, backupSessionRef)
```

Comment:
`BackupSession` should not be owner of anything. Use `BackupConfiguration` instead.